homeassistant/components/insteon/utils.py

```diff
@@ -377,9 +377,7 @@ def print_aldb_to_log(aldb):
         )
         logger.info(log_msg)
 
-
-@callback
-def async_add_insteon_entities(
+async def async_add_insteon_entities(
```

Comment:
There is no need for this. Revert this.

Reply:
When I revert this and run it in the debugger I receive the following error:
```
021-02-01 16:46:27 ERROR (SyncWorker_0) [homeassistant.util.logging] Exception in add_entities when dispatching 'insteon_add_entities_binary_sensor': ({'address': x10c03},)
Traceback (most recent call last):
  File "/workspaces/home-assistant/homeassistant/components/insteon/binary_sensor.py", line 56, in add_entities
    async_add_insteon_entities(
  File "/workspaces/home-assistant/homeassistant/components/insteon/utils.py", line 395, in async_add_insteon_entities
    async_add_entities(new_entities)
  File "/workspaces/home-assistant/homeassistant/helpers/entity_platform.py", line 263, in _async_schedule_add_entities
    task = self.hass.async_create_task(
  File "/workspaces/home-assistant/homeassistant/core.py", line 393, in async_create_task
    task: asyncio.tasks.Task = self.loop.create_task(target)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 431, in create_task
    task = tasks.Task(coro, loop=self, name=name)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 721, in call_soon
    self._check_thread()
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 758, in _check_thread
    raise RuntimeError(
RuntimeError: Non-thread-safe operation invoked on an event loop other than the current one
```

Reply:
I found this error when trying to fix #44848 and I cannot move forward without clearing this runtime error.

Reply:
https://developers.home-assistant.io/docs/asyncio_categorizing_functions

Reply:
I am sure you know this but the error above only shows up in the debugger because `asyncio._check_thread` is only run in debug mode.

Reply:
Read that link I posted and look at my other comment, just tag it with `@callback`.

Reply:
OK, I see the issue now. The `add_entities` methods inside each platform needed to be decorated with `@callback` not just `insteon.utils.async_add_insteon_entities`.  I am going to close this PR and fix it the correct way. Sorry about that.